lib/ws_servers/api/handlers/on_selected_algo_resume_remove.js

```diff
@@ -28,15 +29,18 @@ module.exports = async (server, session, msg) => {
     if (algoWorker) {
       tasks.push(...payload[mode].remove.map(({ gid }) => algoWorker.cancelOrder(gid)))
 
+      const activeAOs = algoWorker.getActiveAlgosFromHost()
+
       for (const { algoID, gid } of payload[mode].resume) {
-        const ao = await AlgoOrder.get({ algoID, gid })
-        const { state, createdAt } = ao || {}
+        const instance = activeAOs.find(({ state }) => state.gid === gid && state.id === algoID)
```

Comment:
Why resume them if they are already running in AO host? If they are already running, skip it and if they are not running, read from DB and resume them. And since, we would be resuming AOs in case of application startup or after network reconnection, I don't think we would have any running AOs anyway. But, yeah it's better to check if they are already running.

Reply:
@avsek477 I am not sure how to check whether the instance is already running? I believe `active` flag in state is sufficient to check whether it is running, right?

Reply:
If getActiveAlgosFromHost() returns the AO then, it is running. But, what I am saying is `getActiveAlgosFromHost` will always return empty array because there won't be any instances running, in case of application start up or after network reconnection. 
P.S. Please always run the application and try the changes after it's developed and check if the intended changes are achieved or not. You will have a clear idea then on how the application works.